s3tests/functional/test_s3.py

```diff
@@ -5709,6 +5709,83 @@ def test_list_multipart_upload():
     upload2.cancel_upload()
     upload3.cancel_upload()
 
+@attr(resource='bucket')
+@attr(method='put')
+@attr(operation='list multi-part uploads')
+def test_list_multipart_upload_many():
+    bucket = get_new_bucket()
+    mb = 1024 * 1024
+    (upload1, data) = _multipart_upload(bucket, "foo", 5 * mb)
+    (upload2, data) = _multipart_upload(bucket, "bar", 5 * mb)
+    (upload3, data) = _multipart_upload(bucket, "baz", 5 * mb)
+    (upload4, data) = _multipart_upload(bucket, "asd", 5 * mb)
+    (upload5, data) = _multipart_upload(bucket, "haha", 5 * mb)
+
+    l = bucket.get_all_multipart_uploads(max_uploads=2)
+    eq(len(l), 2)
+
+    l = bucket.get_all_multipart_uploads(max_uploads=4)
+    eq(len(l), 4)
+ 
+    l = bucket.get_all_multipart_uploads()
+    eq(len(l), 5)
+
+    upload1.cancel_upload()
+    upload2.cancel_upload()
+    upload3.cancel_upload()
+    upload4.cancel_upload()
+    upload5.cancel_upload()
+
+@attr(resource='bucket')
+@attr(method='put')
+@attr(operation='list multi-part uploads prefix')
+def test_list_multipart_upload_prefix():
+    bucket = get_new_bucket()
+    mb = 1024 * 1024
+    (upload1, data) = _multipart_upload(bucket, "photos/2006/January/sample.jpg", 5 * mb)
+    (upload2, data) = _multipart_upload(bucket, "photos/2006/February/sample.jpg", 5 * mb)
+    (upload3, data) = _multipart_upload(bucket, "sample.wmv", 5 * mb)
+    (upload4, data) = _multipart_upload(bucket, "sample.jpg", 5 * mb)
+
+    l = bucket.get_all_multipart_uploads(prefix='photos')
+    l = list(l)
+    eq(len(l), 2)
+
+    index = dict([("photos/2006/January/sample.jpg", 1),
+                  ("photos/2006/February/sample.jpg", 1)])
+    for upload in l:
+        index[upload.key_name] -= 1;
```

Comment:
use `set` instead?

Reply:
It's ok to use set here. Because the  name of two multi_uploads may be same, it will list all of them. So use dict may be better.  I wrote this test for reference to the case of "test_list_multipart_upload".